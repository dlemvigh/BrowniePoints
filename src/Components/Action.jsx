import React from "react";

const Action = ({ name, score, onAddScore }) => (
  <button 
    type="button" 
    onClick={() => onAddScore(score)}
    className={"btn btn-primary"}
  >{score > 0 && "+"}{score} {name}</button>
);

export default Action;