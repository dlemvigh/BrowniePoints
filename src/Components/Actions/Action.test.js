import React from 'react';
import { shallow } from 'enzyme';
import Action from './Action';

it('matches snapshot, positive score', () => {  
  const wrapper = shallow(<Action name="Do dishes" score={10} />);
  expect(wrapper).toMatchSnapshot();
})

it('matches snapshot, negative score', () => {  
  const wrapper = shallow(<Action name="Amusement park" score={-200} />);
  expect(wrapper).toMatchSnapshot();
})

it('triggers callback', () => {
  const score = 1;
  const name = "test";
  const callback = jest.fn();
  const wrapper = shallow(<Action onAddScore={callback} score={score} name={name} />);

  expect(callback.mock.calls.length).toBe(0);

  const button = wrapper.find("button");
  button.simulate("click");

  expect(callback.mock.calls.length).toBe(1);
  expect(callback.mock.calls[0]).toEqual([score, name]);
})