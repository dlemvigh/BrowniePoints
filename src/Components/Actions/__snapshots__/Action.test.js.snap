// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, negative score 1`] = `
<button
  className="btn btn-danger action"
  onClick={[Function]}
  type="button"
>
  -200
   
  Amusement park
</button>
`;

exports[`matches snapshot, positive score 1`] = `
<button
  className="btn btn-success action"
  onClick={[Function]}
  type="button"
>
  +
  10
   
  Do dishes
</button>
`;
